pkg/inventory/policy.go

```diff
@@ -119,7 +121,7 @@ func CanPrune(inv InventoryInfo, obj *unstructured.Unstructured, policy Inventor
 	matchStatus := inventoryIDMatch(inv, obj)
 	switch matchStatus {
 	case Empty:
-		return false
+		return policy == AdoptIfNoInventory || policy == AdoptAll
```

Comment:
This looks good, but should we also consider the `NoMatch` case? It seems to me that if the user provided the `AdoptAll` value, then we should maybe allow pruning in this case. In general, should we treat pruning the same as adoption, so basically if a resource can be updated (and therefore adopted), should we also allow it to be pruned?

Reply:
That's a good point. I'm thinking how we could encounter the case that we need to prune a resource with `AdoptAll` policy and `NoMatch`. Since `AdoptAll` won't be exposed to the cli, so we won't encounter this case through cli. It could only happen for the library consumers.

When the library is called with `AdoptAll` policy and the inventory object is updated. However, a resource is failed to be applied. So the `owning-inventory` annotation doesn't match with the inventory object. Later this resource is removed from the config. So it needs to be pruned in the re-apply. Then we encounter the case mentioned above. In this case, it makes sense to allow pruning the resource.

So I think we should allow the pruning. The resource to be pruned can be viewed as a failed apply + prune.

Reply:
updated.